File: C/Queue/Array_Queue/Linear_Queue/Queue.c
#include "Queue.h"

void Queue_init(Queue *queue, int maxSize)
{
    queue->array = (int *)malloc(sizeof(int) * maxSize);
    queue->capacity = maxSize;
    queue->front = -1;
    queue->rear = -1;
    queue->size = 0;
}

int isEmpty(Queue *queue)
{
    return queue->size == 0;
}

int isFull(Queue *queue)
{
    return (queue->size == queue->capacity);
}

void enqueue(Queue *queue, int value)
{
    if (isFull(queue))
    {
        printf("Queue is full\n");
        return;
    }
    queue->rear++;
    queue->array[queue->rear] = value;
    queue->size++;
}

int dequeue(Queue *queue)
{
    if (isEmpty(queue))
    {
        printf("Queue is Empty");
        return -1;
    }

    int val = queue->array[queue->front];
    queue->front++;
    return val;
}

int peek(Queue *queue)
{
    if (isEmpty(queue))
    {
        printf("Queue is Empty");
        return -1;
    }
    return queue->array[queue->front];
}

File: Java/Queue/Array_Queue/Linear_Queue/Queue.java
package Java.Queue.Array_Queue.Linear_Queue;

public class Queue {
    private int front;
    private int rear;
    private int length;
    private int capacity;
    private int[] que;

    Queue() {
        this.front = -1;
        this.rear = -1;
        this.length = 0;
        this.capacity = 10;
        this.que = new int[10];
    }

    Queue(int capacity) {
        this.front = -1;
        this.rear = -1;
        this.length = 0;
        this.capacity = capacity;
        this.que = new int[capacity];
    }

    public int peek() {
        if (isEmpty()) {
            System.out.println("Queue is Empty");
            return 0;
        }

        return que[front];
    }

    public void enqueue(int value) {
        if (isFull()) {
            System.out.println("Queue is full");
            return;
        }

        if (isEmpty()) {
            front++;
        }

        rear++;
        length++;
        que[rear] = value;
    }

    public void dequeue() {
        if (isEmpty()) {
            System.out.println("Queue is empty");
        } else {
            front++;
            length--;
        }
    }

    public int size() {
        return length;
    }

    public boolean isEmpty() {
        return length == 0;
    }

    public boolean isFull() {
        return capacity == rear - 1;
    }

}
